feat(sources): add custom auth strategy for components with HTTP server (#22236)
vectordotdev/vector: changelog.d/22236_custom_server_auth_strategy.feature.md

> Custom authorization strategy is now supported for sources running
> HTTP servers (`http_server` source, `prometheus` source, `datadog_agent`, etc.).
> If strategy is not explicitly defined, it defaults to `basic`, which is the current behavior.

Reviewer: ```suggestion
Sources running HTTP servers (`http_server` source, `prometheus` source, `datadog_agent`, etc.) now support a new `custom` authorization strategy . If a strategy is not explicitly defined, it defaults to `basic`, which is the current behavior.
```
